lib/spack/llnl/util/filesystem.py

```diff
@@ -918,8 +918,7 @@ def remove_if_dead_link(path):
         path (str): The potential dead link
     """
     if os.path.islink(path):
-        real_path = os.path.realpath(path)
-        if not os.path.exists(real_path):
+        if not os.path.exists(path):
```

Comment:
As a minor improvement, we can collapse the two conditions
```python
if os.path.islink(path) and not os.path.exists(path):
    os.unlink(path)
```

Reply:
done